### PR TITLE
FIX ini sections to also work with initialise

### PR DESCRIPTION
**Pull request by hafriedlander**
Also fixes using the default section if a vault-specific section doesnt exist,
and using ini files without sections

### CODE REVIEW

File: Frisp/FrispEnv.php [hunk 1]
```diff
@@ -7,16 +7,27 @@
...
 	/** @var string */
-	private $vault;
+	private $vaultOrConfigKey;
+
+	function __construct($options = null, $role=null, $bucket=null, $vaultOrConfigKey = null, $parseini = true) {
+
+		$this->inifile = array();
+
+		if ($parseini) {
+			if (file_exists($_SERVER['HOME'].'/.aws/frisp.ini')) {
```
Comment (stojg) [inline Frisp/FrispEnv.php]: can we rely on $_SERVER in a CLI environment?

Comment (stojg) [inline Frisp/FrispEnv.php]: Ah, $ENV gets chucked in to $_SERVER

File: Frisp/FrispEnv.php [hunk 2]
```diff
@@ -7,16 +7,27 @@
 class FrispEnv {
 
 	/** @var array */
+	private $inifile;
```
Comment (stojg) [inline Frisp/FrispEnv.php]: minor: perhaps better named as $iniSettings.. I thought it was the file path..

File: Frisp/FrispEnv.php [hunk 3]
```diff
@@ -7,16 +7,27 @@
...
+	/** @var array */
 	private $options;
 	/** @var string */
 	private $bucket;
 	/** @var string */
-	private $vault;
+	private $vaultOrConfigKey;
```
Comment (stojg) [inline Frisp/FrispEnv.php]: minor: not optimal to have a variable name with an `or` in it, code smell.

File: Frisp/FrispEnv.php [hunk 4]
```diff
@@ -25,9 +36,32 @@ function __construct($options = null, $role=null, $bucket=null, $vault = null) {
 		}
 	}
 
+	function getOptions() {
+		if ($this->vaultOrConfigKey && isset($this->inifile[$this->vaultOrConfigKey])) {
```
Comment (stojg) [inline Frisp/FrispEnv.php]: Question: should `isset()` be replaced with an `!empty()`, what is the expected behaviour?

File: Frisp/FrispEnv.php [hunk 5]
```diff
@@ -25,9 +36,32 @@ function __construct($options = null, $role=null, $bucket=null, $vault = null) {
...
+			$baseOptions = $this->inifile[$this->vaultOrConfigKey];
+		} elseif (isset($this->inifile['default'])) {
```
Comment (stojg) [inline Frisp/FrispEnv.php]: same here as above (isset vs !empty)

File: Frisp/FrispEnv.php (hunk 1 above)
Comment (hafriedlander) [inline Frisp/FrispEnv.php]: Yeah, I think HOME is pretty reliable. The PHP CLI execution environment definitely shows that it's come second after the web execution environment was already old.

File: Frisp/FrispEnv.php (hunk 4 above)
Comment (hafriedlander) [inline Frisp/FrispEnv.php]: I think if you had an empty vault-specific config section, you're saying "don't use the defaults", so isset is right.

**Comment by hafriedlander** [comment]
Actually, I've decided I don't like overloading vault & config key, given it won't affect initialise anyway.

**Pull request closed without merging**